ext/storage_sources/azure_store/azure_connection.cpp

```diff
@@ -95,3 +102,56 @@ azure_connection::get_object(const std::string &path) const
 {
     return 0;
 }
+
+int
+azure_connection::object_exists(const std::string &object_key, bool &exists) const
+{
+    exists = false;
+    std::string obj = _object_prefix + object_key;
+
+    auto list_blob_response = _azure_client.ListBlobs();
+
+    int ret = 0;
+
+    for (const auto blob_item : list_blob_response.Blobs) {
+        // Check if object exists.
+        if (blob_item.Name.compare(obj) == 0) {
+            exists = true;
+            // Check if object is deleted and has not been cleared by garbage collection.
+            if (blob_item.IsDeleted) {
+                exists = false;
+                ret = -1;
```

Comment:
np: can we just return -1 here instead of assigning ret = -1 and returning it later?

Reply:
yep that sounds good, done :)